src/components/ConfigEditor/ConnectionConfig.tsx

```diff
@@ -143,6 +146,30 @@ const ConnectionConfig: React.FC<ConnectionConfigProps> = ({
         isConfigured={!!secureJsonFields?.clientSecret}
         tooltip={clientSecretTooltip}
       />
+
+      <InlineField
+        label="Use On-Behalf-Of"
+        labelWidth={26}
+        tooltip={
+          <>
+            Propagate Grafana client credentials to ADX with a token exchange. When enabled the service account (Client
+            ID) impersonates the user by augmenting the access token. See the{' '}
+            <a
+              target="_blank"
+              href="https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-on-behalf-of-flow"
+            >
+              developer documentation
+            </a>{' '}
+            for detail on the concept.
```

Comment:
I would add a note that OAuth should be enabled and that it requires Grafana > 8.3.3

Reply:
Right. I wanted to verify the configuration and version in the plugin user-interface but got stuck while trying. Is the main configuration available in the plugin somehow?

Reply:
It's possible to retrieve that configuration and hide this switch but probably is not worthy. The note explaining it should be enough.

Reply:
What do you mean by not worthy? My time is free to you. 😉 ASML gladly pays to have this plugin inform the user, rather than shift responsibility with an increasing list of warnings and instructions.

Reply:
I mean maintaining (not only implementing) the code that retrieves the configuration, checks and compares the version and enables/disables this input. The notice with the version should be enough.